### PR TITLE
fix: Crash when bluetooth is off and Send button is clicked

### LINKED ISSUE #38: Crash when bluetooth is off and Send button is clicked
**Describe the bug**
Bluetooth adapter is not turned on error is showing up as there's no check in the code.

**To Reproduce**
Turn off bluetooth and click send button.

**Expected behavior**
App should not crash. 

**Logs**
AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nilhcem.blenamebadge, PID: 15932
    java.lang.IllegalStateException: BT Adapter is not turned ON
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeUtils.checkAdapterStateOn(BluetoothLeUtils.java:136)
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerImplLollipop.startScanInternal(BluetoothLeScannerImplLollipop.java:57)
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerCompat.startScan(BluetoothLeScannerCompat.java:112)
        at com.nilhcem.blenamebadge.device.bluetooth.ScanHelper.startLeScan(ScanHelper.kt:57)
        at com.nilhcem.blenamebadge.ui.message.MessagePresenter.sendBytes(MessagePresenter.kt:36)
        at com.nilhcem.blenamebadge.ui.message.MessagePresenter.sendBitmap(MessagePresenter.kt:25)
        at com.nilhcem.blenamebadge.ui.message.MessageActivity$onCreate$3.onClick(MessageActivity.kt:69)
        at android.view.View.performClick(View.java:6291)
        at android.view.View$PerformClick.run(View.java:24931)
        at android.os.Handler.handleCallback(Handler.java:808)
        at android.os.Handler.dispatchMessage(Handler.java:101)
        at android.os.Looper.loop(Looper.java:166)
        at android.app.ActivityThread.main(ActivityThread.java:7529)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)

**Smartphone Info:**

|Device              | Huawei Honor 7x |
|Android Version| Oreo 8.1    |

**Additional context**
Need to check if the bluetooth is enabled before starting the scan.

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by balaji-ramavathu**
Fixes #38

Changes: A check to know if bluetooth is enabled is added before the main logic starts in send button onClick method and a toast is displayed if blutooth is not enabled.

### CODE REVIEW

**Comment by balaji-ramavathu** [comment]
@iamareebjamal you once reviewed my previous PR for this issue. Can you just check this once and merge it if everything is good ? :-)

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 1]
```diff
@@ -57,21 +57,25 @@ class MessageActivity : AppCompatActivity() {
...
+                    presenter.sendBitmap(this, BitmapFactory.decodeResource(resources, R.drawable.mix2))
+                } else {
+                    presenter.sendMessage(this, convertToDeviceDataModel())
                 }
-            }, SCAN_TIMEOUT_MS)
-            if (content.text.isEmpty()) {
-                presenter.sendBitmap(this, BitmapFactory.decodeResource(resources, R.drawable.mix2))
             } else {
-                presenter.sendMessage(this, convertToDeviceDataModel())
+                Toast.makeText(this, getString(R.string.txt_turn_on_bluetooth), Toast.LENGTH_LONG).show()
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Don't use android classes in presenter. Use view interface

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 1 above)
Comment (balaji-ramavathu) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Sorry I didn't get it. Which class are you referring to ?

Comment (iamareebjamal) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Toast

Comment (balaji-ramavathu) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: I've made the asked changes.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessagePresenter.kt [hunk 2]
```diff
@@ -50,4 +51,8 @@ class MessagePresenter {
             }
         }
     }
+
+    fun showToast(context: Context, message: String) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessagePresenter.kt]: Don't use toast in presenter. I mistook activity for presenter. Revert to previous behaviour

Comment (balaji-ramavathu) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessagePresenter.kt]: I've reverted it to previous code. Now please merge it.

**Review by iamareebjamal — APPROVED** [review]

**Comment by balaji-ramavathu** [comment]
@iamareebjamal can you merge this ?

**Pull request merged**